packages/grpc-native-core/src/server.js

```diff
@@ -509,6 +509,46 @@ ServerReadableStream.prototype.getPeer = getPeer;
 ServerWritableStream.prototype.getPeer = getPeer;
 ServerDuplexStream.prototype.getPeer = getPeer;
 
+/**
+ * Get the authentication context this call/stream currently has.
+ * @alias grpc~ServerUnaryCall#getAuthContext
+ * @return {(Object|Null)} The list of authentication properties
+ */
+function getAuthContextSingleton(property) {
+  if (property.value.length !== 1) {
+    throw new Error('property ' + property.name + ' isn\'t a singleton');
+  }
+  return { name: property.name, value: property.value[0] };
+}
+
+const getAuthContextMapping = {
+  transportSecurityType: getAuthContextSingleton,
+  x509PemCert: function(property) {
+    property = getAuthContextSingleton(property);
+    var cert = common.pemCertificateToRaw(property.value);
+    if (cert instanceof Error) throw cert;
+    return { name: 'sslPeerCertificate', value: { raw: cert } };
+  }
+}
+
+function getAuthContext() {
+  /* jshint validthis: true */
+  var authContext = this.call.getAuthContext();
+  return _.reduce(authContext, function(obj, value, key) {
+    key = _.camelCase(key);
```

Comment:
This doesn't really seem necessary. The `authContextMapping` can just use the original names as keys.

Reply:
Myeah, this is a remnant of what I did previously where THIS was the sole transformation. Still, this kind of has the nice property of letting us have a `getAuthContextMapping` that has camelCase as keys as with the rest of the surrounding code, so it makes my eyes twitch a bit less when reading it, but if you feel strongly about it I'll remove it.